### PR TITLE
cat-manifest: added implementation

### PR DESCRIPTION
**Pull request by dgonyeo**
This commit adds a command to cat the manifest, either from the current
build or from a given aci.

### CODE REVIEW

**Comment by jonboulle** [comment]
Hmm. This feels a little weird in `acbuild`; displaying the manifest of the current build makes sense, but of a given ACI is a bit odd. Perhaps this functionality should just stay in actool?

**Comment by dgonyeo** [comment]
I added the `--aci` flag to prevent people from doing `acbuild --modify=some.aci cat-manifest`, since that would result in expanding the entire ACI to read a few hundred bytes of a file.

I'd be fine with removing the `--aci` flag, but if I do should I allow `cat-manifest` to be used with the `--modify` flag?

**Comment by jonboulle** [comment]
Oh right. Hmm. I think for now, either just disallow it with --modify, or
have that codepath do the same thing you're doing now with lib.CatManifest.
Essentially --modify is --in-place or --act-on whatever.

**Comment by dgonyeo** [comment]
I chose to have `--modify` use the same code path as was being used in `runCat`. How's that look?

**Comment by jonboulle** [comment]
Better, but I really don't like more and more getting shoehorned into runWrapper :-). Can we split that out?

**Comment by dgonyeo** [comment]
Moved to `acbuild/cat-manifest.go`

File: acbuild/acbuild.go [hunk 1]
```diff
@@ -126,14 +126,17 @@ func runWrapper(cf func(cmd *cobra.Command, args []string) (exit int)) func(cmd
...
+		switch command {
+		case "cat-manifest":
+			cmdExitCode = runCatOnACI(aciToModify)
+			return
+		case "begin", "write", "end", "version":
+			stderr("Can't use the --modify flag with %s.", command)
+			cmdExitCode = 1
+			return
+		default:
+			break
```
Comment (jonboulle) [inline acbuild/acbuild.go]: Remove this.

File: acbuild/cat-manifest.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+func runCatOnACI(aciToModify string) (exit int) {
+	if len(args) != 0 {
+		cmdCat.Usage()
+		return 1
+	}
+	err := lib.CatManifest(aciToModify, prettyPrint)
+	if err != nil {
+		stderr("cat-manifest: %v", err)
+		cmdExitCode = 1
```
Comment (jonboulle) [inline acbuild/cat-manifest.go]: Nack, this should be returned, not set...

File: util/manifest.go [hunk 3]
```diff
@@ -75,3 +77,23 @@ func ModifyManifest(fn func(*schema.ImageManifest), acipath string) error {
 
 	return nil
 }
+
+// PrintManifest will print the given manifest to stdout, optionally inserting
+// whitespace to make it more human readable.
+func PrintManifest(man *schema.ImageManifest, prettyPrint bool) error {
+	if !prettyPrint {
```
Comment (jonboulle) [inline util/manifest.go]: var manblob, err and then you can share code for most of these paths

**Comment by dgonyeo** [comment]
Should be ready for another pass

File: lib/cat-manifest.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+	tr, err := aci.NewCompressedTarReader(file)
+	if err != nil {
+		return fmt.Errorf("error decompressing image: %v", err)
+	}
+	defer tr.Close()
+
+	for {
+		hdr, err := tr.Next()
+		switch {
```
Comment (jonboulle) [inline lib/cat-manifest.go]: if err != nil but != io.EOF could we get stuck in here?

**Comment by jonboulle** [comment]
Nice, one question, lgtm after that

File: lib/cat-manifest.go (hunk 4 above)
Comment (dgonyeo) [inline lib/cat-manifest.go]: I suppose we could.

Added a 
```
case err != nil:
    return err
```

**Pull request merged**